### PR TITLE
Add random handle support to directory and matrix fuzz tests 

### PR DESCRIPTION
**Pull request by jzaffiro**
Since support for adding handles into DDS content in the fuzz tests is now enabled, update the directory and matrix fuzz tests to include handles randomly in the content they generate. 

[AB#7499](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7499), [AB#7500](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7500)

### CODE REVIEW

File: packages/dds/map/src/test/mocha/directoryEquivalenceUtils.ts [hunk 1]
```diff
@@ -28,14 +35,40 @@ function assertEventualConsistencyCore(
...
-			first.get(key),
-			second.get(key),
-			`Key not found or value not matching ` +
-				`key: ${key}, value in dir first at path ${first.absolutePath}: ${first.get(
-					key,
-				)} and in second at path ${second.absolutePath}: ${second.get(key)}`,
-		);
+		const firstVal: unknown = first.get(key);
+		const secondVal: unknown = second.get(key);
+		const firstObj: FluidObject<IFluidHandle> = firstVal as FluidObject<IFluidHandle>;
```
Comment (anthony-murphy) [inline packages/dds/map/src/test/mocha/directoryEquivalenceUtils.ts:40]: this should be very similar to the map function. the `as FluidObject<IFluidHandle>` should be unnecessary if you check if they are objects first

File: packages/dds/matrix/src/test/matrix.fuzz.spec.ts [hunk 2]
```diff
@@ -89,18 +91,37 @@ class TypedMatrixFactory extends SharedMatrixFactory {
...
 		for (let col = 0; col < a.colCount; col++) {
 			const aVal = a.getCell(row, col);
 			const bVal = b.getCell(row, col);
-			assert.equal(
-				aVal,
-				bVal,
-				`${a.id} and ${b.id} differ at (${row}, ${col}): ${aVal} vs ${bVal}`,
-			);
+			if (
+				aVal !== null &&
```
Comment (anthony-murphy) [inline packages/dds/matrix/src/test/matrix.fuzz.spec.ts:102]: us isObject from import { isObject } from "@fluidframework/core-utils/internal";

File: packages/dds/matrix/src/test/matrix.fuzz.spec.ts [hunk 3]
```diff
@@ -89,18 +91,37 @@ class TypedMatrixFactory extends SharedMatrixFactory {
...
+				typeof aVal === "object" &&
+				bVal !== null &&
+				typeof bVal === "object"
+			) {
+				const aObj: FluidObject<IFluidHandle> = aVal as FluidObject<IFluidHandle>;
+				const bObj: FluidObject<IFluidHandle> = bVal as FluidObject<IFluidHandle>;
+				const aHandle = aObj.IFluidHandle ? aObj.IFluidHandle?.get() : aObj;
+				const bHandle = bObj.IFluidHandle ? bObj.IFluidHandle?.get() : bObj;
```
Comment (anthony-murphy) [inline packages/dds/matrix/src/test/matrix.fuzz.spec.ts:110]: if (isObject(aVal)) {
			const aObj: FluidObject<IFluidHandle> = aVal;
			assert(
				isObject(bVal),
				`${a.id} and ${b.id} differ at ${key}: a is an object, b is not`,
			);
			const bObj: FluidObject<IFluidHandle> = bVal;
			const aHandle = aObj.IFluidHandle ? await aObj.IFluidHandle?.get() : aObj;
			const bHandle = bObj.IFluidHandle ? await bObj.IFluidHandle?.get() : bObj;

Comment (anthony-murphy) [reply packages/dds/matrix/src/test/matrix.fuzz.spec.ts:110]: this pattern can be reused in map an dir as well

File: packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts [hunk 4]
```diff
@@ -339,6 +344,7 @@ describe("SharedDirectory fuzz Create/Delete concentrated", () => {
 		validationStrategy: { type: "fixedInterval", interval: defaultOptions.validateInterval },
 		reconnectProbability: 0.15,
 		numberOfClients: 3,
+		handleGenerationDisabled: true,
```
Comment (anthony-murphy) [inline packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts:347]: can you add a comment about why it is disable in these tests

File: packages/dds/map/src/test/mocha/directoryEquivalenceUtils.ts (hunk 1 above)
Comment (jzaffiro) [reply packages/dds/map/src/test/mocha/directoryEquivalenceUtils.ts]: I still got lint errors without the cast after checking that they are objects, so all of the values are still casted

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**